diff --git a/target/linux/ath79/dts/qca9531_embstar_mp05.dts b/target/linux/ath79/dts/qca9531_embstar_mp05.dts
new file mode 100644
index 0000000..a393304
--- /dev/null
+++ b/target/linux/ath79/dts/qca9531_embstar_mp05.dts
@@ -0,0 +1,162 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca953x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	compatible = "embstar,mp05", "qca,qca9531";
+	model = "Embstar MP05";
+
+	aliases {
+		label-mac-device = &eth1;
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		pinctrl-names = "default";
+		pinctrl-0 = <&jtag_disable_pins>;
+
+		reset {
+			linux,code = <KEY_RESTART>;
+			label = "reset";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			function = "LED_FUNCTION_POWER";
+			color = "<LED_COLOR_ID_RED>";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		lan1 {
+			label = "green:lan1";
+                        gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+                };
+
+                lan2 {
+			label = "green:lan2";
+                        gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+                };
+
+                lan3 {
+			label = "green:lan3";
+                        gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+                };
+
+                lan4 {
+			label = "green:lan4";
+                        gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+                };
+
+                wan {
+			function = LED_FUNCTION_WAN;
+			color = <LED_COLOR_ID_GREEN>;
+                        gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+                };
+
+		wlan {
+			function = LED_FUNCTION_WLAN;
+                        color = <LED_COLOR_ID_AMBER>;
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+	};
+};
+
+&pcie0 {
+	status = "okay";
+};
+
+&usb0 {
+	status = "okay";
+};
+
+&usb_phy {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x040000 0x010000>;
+				read-only;
+			};
+			
+			partition@50000 {
+                                compatible = "denx,uimage";
+                                label = "firmware";
+                                /*reg = <0x050000 0x1ef0000>;*/
+                                reg = <0x050000 0xfa0000>;
+                        };
+			
+			art: partition@ff0000 {
+                                label = "art";
+                                reg = <0xff0000 0x010000>;
+                                read-only;
+
+				nvmem-layout {
+					compatible = "fixed-layout";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					macaddr_art_0: macaddr@0 {
+						reg = <0x0 0x6>;
+					};
+
+					macaddr_art_6: macaddr@6 {
+						reg = <0x6 0x6>;
+					};
+				};
+                        };
+		};
+	};
+};
+
+&eth0 {
+	status = "okay";
+	phy-handle = <&swphy4>;
+	nvmem-cells = <&macaddr_art_0>;
+	nvmem-cell-names = "mac-address";
+};
+
+&eth1 {
+	nvmem-cells = <&macaddr_art_6>;
+	nvmem-cell-names = "mac-address";
+};
+
+/*WMAC disabled for now*/
+&wmac {
+	status = "disabled";
+	mtd-cal-data = <&art 0x1000>;
+};
diff --git a/target/linux/ath79/generic/base-files/etc/board.d/01_leds b/target/linux/ath79/generic/base-files/etc/board.d/01_leds
index 525224b..1364c08 100644
--- a/target/linux/ath79/generic/base-files/etc/board.d/01_leds
+++ b/target/linux/ath79/generic/base-files/etc/board.d/01_leds
@@ -255,6 +255,14 @@ dlink,dir-859-a1)
 dlink,dir-859-a3)
 	ucidef_set_led_switch "wan" "WAN" "green:wan" "switch0" "0x20"
 	;;
+embstar,mp05)
+	ucidef_set_led_wlan "wlan" "WLAN" "amber:wlan" "phy0tpt"
+	ucidef_set_led_netdev "wan" "LAN" "green:wan" "eth1"
+	ucidef_set_led_switch "lan1" "LAN1" "green:lan1" "switch0" "0x02"
+	ucidef_set_led_switch "lan2" "LAN2" "green:lan2" "switch0" "0x04"
+	ucidef_set_led_switch "lan3" "LAN3" "green:lan3" "switch0" "0x08"
+	ucidef_set_led_switch "lan4" "LAN4" "green:lan4" "switch0" "0x10"
+	;;
 engenius,ens202ext-v1|\
 engenius,enstationac-v1)
 	ucidef_set_rssimon "wlan0" "200000" "1"
diff --git a/target/linux/ath79/generic/base-files/etc/board.d/02_network b/target/linux/ath79/generic/base-files/etc/board.d/02_network
index 6c1cb86..af09b49 100644
--- a/target/linux/ath79/generic/base-files/etc/board.d/02_network
+++ b/target/linux/ath79/generic/base-files/etc/board.d/02_network
@@ -346,6 +346,11 @@ ath79_setup_interfaces()
 		ucidef_add_switch "switch0" \
 			"0@eth0" "1:wan" "2:lan:3" "3:lan:2"
 		;;
+	embstar,mp05)
+		ucidef_set_interface_lan "eth1"
+		ucidef_add_switch "switch0" \
+			"0@eth0" "1:lan:1" "2:lan:2" "3:lan:3" "4:lan:4"
+		;;
 	engenius,eap300-v2)
 		ucidef_add_switch "switch0" \
 			"0@eth0" "1:lan"
diff --git a/target/linux/ath79/image/generic.mk b/target/linux/ath79/image/generic.mk
index e7960ef..77d6116 100644
--- a/target/linux/ath79/image/generic.mk
+++ b/target/linux/ath79/image/generic.mk
@@ -1412,6 +1412,16 @@ define Device/embeddedwireless_dorin
 endef
 TARGET_DEVICES += embeddedwireless_dorin
 
+define Device/embstar_mp05
+  SOC := qca9531
+  DEVICE_VENDOR := Embstar
+  DEVICE_MODEL := MP05
+  DEVICE_PACKAGES := kmod-usb2 kmod-ath10k-ct ath10k-firmware-qca9887-ct luci
+  IMAGE_SIZE := 16000k
+  SUPPORTED_DEVICES += embstar
+endef
+TARGET_DEVICES += embstar_mp05
+
 define Device/engenius_eap1200h
   $(Device/senao_loader_okli)
   SOC := qca9557
